### ISSUE TITLE
update from version 1.31 to version 1.32 throws on all tests NoClassDefFound javax/persistence/Entity

### ISSUE THREAD

**Issue by avaysberg** [jmockit/jmockit1#438]
Please provide the following information:

* Version of JMockit that was used: 
1.32
* Description of the problem:
if I updated to use version 1.32 or high i receive Exception NoClassDefFound and if i move back to 1.31 it works fine:

`TestPaging.testGetSize » NoClassDefFound javax/persistence/Entity`

Should i have this class  javax/persistence/Entity for Tests? We don't use the java persistence.

Regards

**Comment by rliesenfeld** [comment]
How would I reproduce the problem? There are two sample modules in the JMockit project (java8testing, LoginService) which don't have `javax.persistence` in the classpath, and they don't throw any `NoClassDefFoundError`s.

**Comment by avaysberg** [comment]
HI,
excuse me, but i don't have time, today i prepared maven test project and reproduce the problem. It happens, some art of dependencies (Tomcat embedded) in class path:
```
<dependencies>
        <dependency>
            <groupId>org.testng</groupId>
            <artifactId>testng</artifactId>
            <version>6.9.9</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.jmockit</groupId>
            <artifactId>jmockit</artifactId>
            <version>1.32</version>
        </dependency>
        <dependency>
            <groupId>javax.servlet</groupId>
            <artifactId>javax.servlet-api</artifactId>
            <version>3.1.0</version>
        </dependency>
        <dependency>
            <groupId>org.apache.tomcat.embed</groupId>
            <artifactId>tomcat-embed-jasper</artifactId>
            <version>7.0.47</version>
            <scope>runtime</scope>
        </dependency>
        <dependency>
            <groupId>org.apache.tomcat.embed</groupId>
            <artifactId>tomcat-embed-core</artifactId>
            <version>7.0.47</version>
        </dependency>
        <dependency>
            <groupId>org.apache.tomcat.embed</groupId>
            <artifactId>tomcat-embed-logging-juli</artifactId>
            <version>7.0.47</version>
        </dependency>
        <dependency>
            <groupId>org.apache.tomcat.embed</groupId>
            <artifactId>tomcat-embed-logging-log4j</artifactId>
            <version>7.0.47</version>
        </dependency>
    </dependencies>
```
  Than if i change the jmockit to v. 1.31 it works, but if I change it to 1.32 than it throws the exception bei the call the class TestMockupService 
```
public class TestMockupService {
    @Injectable
    private String host = "test";
    @Injectable
    private int port = 7778;
    @Tested
    private MockingService mockingService;


    @Test
    public void testSendTest(@Mocked OutputStream outputStream) throws Exception {
        MockSocket mockSocket = new MockSocket();
        mockSocket.setOutputStream(outputStream);
        Assert.assertTrue(mockingService.sendTest());
    }

    /**
     * Test method for {@link testng.mocking.MockingService#sendUserName()}.
     */
    @Test
    public void testSendUserName(@Mocked OutputStream outputStream) throws Exception {
        MockSocket mockSocket = new MockSocket();
        mockSocket.setOutputStream(outputStream);
        Assert.assertTrue(mockingService.sendUserName());
    }
}



```
[bug-update-from131-132.zip](https://github.com/jmockit/jmockit1/files/1220508/bug-update-from131-132.zip)

**Comment by schaarda** [comment]
I have the same problem. We have a project with Tomcat and Spring 4.2.3 and I'm totally new to JMockit and wanted to try it on that project. Database-work is done via Spring-Jdbc, not JPA.
I get the same NoClassDefFoundError when I try to run a Test with the JMockit-Runner. I tried downgrading JMockit vom 1.33 to 1.31 and that version works.
Although that's not a satisfying solution.

**Comment by rliesenfeld** [comment]
Fixed in #442.
